### ISSUE TITLE
feat: Implement support for page meta query params in page Url

### ISSUE THREAD

**Issue by OgDev-01** [open-sauced/app#712]
> Can we add the page number to the URL?
> `https://deploy-preview-708--oss-insights.netlify.app/javascript/repositories/filter/top-100-repos?page=4`

This is not present on main, but would make it easier for the user to navigate

Sure, this is a better approach to pagination in general. we could also move other page meta queries to the URL query params as well. Example: `https://deploy-preview-708--oss-insights.netlify.app/javascript/repositories/filter/top-100-repos?page=4&limit=10`

would suggest we create a follow-up issue for this.

_Originally posted by @SunGoldTech in https://github.com/open-sauced/insights/issues/708#issuecomment-1362887074_

**Comment by OgDev-01** [comment]
I'll add this to my list of TODO's for this sprint

**Comment by nickytonline** [comment]
Not only is it easier to navigate and being able to bookmark stuff, it will remove the need for a bunch of use state as the query string is the OG state management.

**Comment by brandonroberts** [comment]
I believe this work has been done with `range` and `limit`, but not `page` yet

**Comment by OgDev-01** [comment]
Yes @brandonroberts , happy to take this on and add page query to it

**Comment by brandonroberts** [comment]
Ok @OgDev-01
